spec/models/comment_spec.rb

```diff
@@ -0,0 +1,24 @@
+# spec/models/comment_spec.rb
+
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
+  describe '#update_post_comments_counter' do
+    it 'increments the associated post comments_counter' do
+      # Arrange
+      user = User.create(name: 'John')
+      post = Post.create(title: 'Hello', author: user)
+      comment = Comment.new(author: user, text: 'This is a comment')
+
+      # Act
+      comment.post = post
+      comment.save
+
+      # Reload the post from the database to get the latest data
+      post.reload
+
```

Comment:
* Hello @Simpleshaikh1 👋 i noticed that your tests are still failing 
![image](https://github.com/Simpleshaikh1/blog-app/assets/52497246/be5f1647-e58e-4db6-a6ec-36ccadd7fc65)
 * As you can tell the error is coming from the association on your schema, on your schema you have user_id but on your test you say author, kindly correct these differences in the association name and your test should all pass 🙏